apps/chat2/chat2.nim

```diff
@@ -21,16 +21,17 @@ import libp2p/[switch,                   # manage transports, a single entry poi
                protocols/secure/secio,   # define the protocol of secure input / output, allows encrypted communication that uses public keys to validate signed messages instead of a certificate authority like in TLS
```

Comment:
Some context/background: _I want to decouple the `wkaunode2` binary from the libraries._

This is the best name I came up with that covers the production binaries/fully fledged apps like `wakunode2` or the `wakubridge`. I am open to suggestions. BTW I discarded the following for the same reason: `binaries`, `bin`, and `cmd`.

Regarding `chat2`... Well, I don't know how to categorize it. But maybe we can use it to show how things should not be done 😆

Reply:
Okay, so as I see this comment is related to my other comment where I ask about prod code depending on `chat2` "example". What do you think if we:
* fix the `chat2` [issue](https://github.com/status-im/nwaku/pull/1228#discussion_r987802325), moving the production code to somewhere in `waku`.
* then in `chat2` import that code, just like the rest of the `waku` imports.
* leave `examples` folder as it it, since i think its very important to keep that naming. i.e. [nim-libp2p](https://github.com/status-im/nim-libp2p/blob/unstable/examples/) also has it.

with this we are addressing your concern of "I want to decouple the wkaunode2 binary from the libraries." right?

Reply:
It is not easy fix, since it implies RLN module code changes. The circular dependency issue is tracked here: https://github.com/status-im/nwaku/issues/1222

We should tackle that in a separate work package. This is one of the reasons why I was asking for some bandwidth to shape the waku RLN module code in this morning meeting 😄

Reply:
> with this we are addressing your concern of "I want to decouple the wkaunode2 binary from the libraries." right?

No... That's a bigger work package. Currently, the `wakunode2.nim` file has two parts:

* The node main routine (that does the setup of the storage, and spins up the different protocols and many other things).
* The waku node type definitions and associated procedures.

This "unit" is too big and does too many things. From a maintenance and debugging point of view, it has been proven that we need to simplify it. How? That's another, longer, discussion.